src/constraint/CollisionFree.cpp

```diff
@@ -27,35 +27,74 @@ statespace::StateSpacePtr CollisionFree::getStateSpace() const
 
 //==============================================================================
 bool CollisionFree::isSatisfied(
-    const aikido::statespace::StateSpace::State* _state) const
+    const aikido::statespace::StateSpace::State* _state,
+    TestableOutcome* outcome) const
 {
+  CollisionFreeOutcome* collisionFreeOutcome = nullptr;
+  if (outcome)
+  {
+    collisionFreeOutcome = dynamic_cast<CollisionFreeOutcome*>(outcome);
+    if (!collisionFreeOutcome)
+      throw std::invalid_argument(
+          "TestableOutcome pointer is not of type CollisionFreeOutcome.");
+    // Clear the outcome object in case it's being reused.
+    collisionFreeOutcome->clear();
+  }
+
   auto skelStatePtr = static_cast<const aikido::statespace::dart::
                                       MetaSkeletonStateSpace::State*>(_state);
   mStatespace->setState(skelStatePtr);
 
   bool collision = false;
   dart::collision::CollisionResult collisionResult;
-  for (auto groups : mGroupsToPairwiseCheck)
+  for (const auto& groups : mGroupsToPairwiseCheck)
   {
     collision = mCollisionDetector->collide(
         groups.first.get(),
         groups.second.get(),
         mCollisionOptions,
         &collisionResult);
+
     if (collision)
+    {
+      if (collisionFreeOutcome)
+      {
+        const auto& contacts = collisionResult.getContacts();
+        for (const auto& elem : contacts)
+        {
+          collisionFreeOutcome->markPairwiseContact(elem);
+        }
```

Comment:
Nit: If you make `CollisionFreeOutcome` a `friend class`, then this could be reduced to:
```c++
collisionFreeOutcome->mContacts = collisionResult.getContacts();
```

Reply:
Thanks for all the awesome feedback! :smile: I had a question about this one- when using  Eigen::aligned_allocator (as mentioned in a previous comment), wouldn't this be not possible? The two vectors appear to use different allocators.

Reply:
I suspect this is actually a bug in DART! :grimacing: @mxgrey @jslee02 

In any case, you can still do the assignment with a bit more boilerplate:
```c++
const auto& contacts = collisionResult.getContacts();
collisionFreeOutcome.assign(std::begin(contacts), std::end(contacts));
```